### ISSUE TITLE
Limit IAM policies

### ISSUE THREAD

**Issue by keyvaann** [RADAR-base/RADAR-K8s-Infrastructure#38]
It seems like `ecr_access` and `ecr_pull_through_cache` policies allow access to all ECR repositories. I think it's safer to limit them to specific resources. Also, I don't see an ECR repository to be defined in the Terraform code, so I'm not why it's being defined.

**Comment by baixiac** [comment]
It was added for the plan to host RADAR images in ECR in order to get around the issues of rate limiting by DockerHub. Maybe there is a general/not-cloud-specific solution now and if so, these policies can be removed.

**Comment by keyvaann** [comment]
It's also good to add the ECR pull through cache to our Terraform code since the issue is still there. I'm making changes to Helm charts to make it easier to define an alternative registry. RADAR-base/radar-helm-charts#310

**Comment by baixiac** [comment]
I have a GH workflow set up (currently disabled to minimise the cost) to sync ECR images with RADAR DockerHub images. Let me know if you guys are interested.

**Comment by keyvaann** [comment]
I haven't used ECR pull through cache yet but I think it is not needed to mirror images? My impression is that you change the image registry in Helm chart and then it should get the image first time from Dockerhub and next times from the cache.
https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache-working-pulling.html

**Comment by baixiac** [comment]
The ECR pull-through cache only supports "official" public DockerHub images and https://hub.docker.com/r/radarbase is not one of them (as of the last time I checked at least).

**Comment by keyvaann** [comment]
It looks like you can use regular images as well:
> For all other Docker Hub images:
> `docker pull aws_account_id.dkr.ecr.region.amazonaws.com/docker-hub/repository_name/image_name:tag`

**Comment by baixiac** [comment]
Can you test that in your own account and confirm if images from the radarbase community organisation can be cached?

**Comment by keyvaann** [comment]
Sure

**Comment by keyvaann** [comment]
The pull through cache works with custom images as well.
```
  Normal  Scheduled  88s   default-scheduler  Successfully assigned default/radar-jdbc-connector-5c9cdf46fd-x8sgx to ip-10-0-23-53.eu-west-2.compute.internal
  Normal  Pulling    87s   kubelet            Pulling image "665845844407.dkr.ecr.eu-west-2.amazonaws.com/docker-hub/radarbase/radar-jdbc-connector:10.5.2"
  Normal  Pulled     62s   kubelet            Successfully pulled image "665845844407.dkr.ecr.eu-west-2.amazonaws.com/docker-hub/radarbase/radar-jdbc-connector:10.5.2" in 24.973s (24.973s including waiting). Image size: 1084429937 bytes.
```

**Comment by baixiac** [comment]
Thanks for testing it. This means ECR has relaxed their rule since this [announcement](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-ecr-pull-through-cache-additional-upstream-registries/). When last time I checked their PTC only supports [officials](https://hub.docker.com/search?badges=official).

**Comment by keyvaann** [comment]
Would it make sense to add the terraform code to create a ECR pull through cache to this repository?

**Comment by baixiac** [comment]
Yes, will TAL. Looks like ECR PTC requires the DockerHub user credentials to be set as an SM secret, which is a less straightforward solution than I thought it would be.

**Comment by baixiac** [comment]
PTAL at https://github.com/RADAR-base/RADAR-K8s-Infrastructure/pull/44, @keyvaann.
